### PR TITLE
Fixed bugs discovered in internal testing

### PR DESCRIPTION
**Pull request by michellexliu**
## Description
1. Previously dropped students can now be undropped if they are added back to the course via another roster upload
2. Fixed unintelligible error message “wrong number of arguments (given 2, expected 0..1)” that appears when new student gets created (incl. blank emails)

## How Has This Been Tested?
See internal testing spreadsheet

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://github.com/autolab/docs)
- [ ] I have updated the documentation accordingly, included in this PR

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -141,8 +141,8 @@ def self.roster_create(email, first_name, last_name, school, major, year)
     user.password_confirmation = temp_pass
     user.skip_confirmation!
 
-    Rails.logger.debug("user email: ", user.email)
-    Rails.logger.debug("user pswd: ", user.password)
+    # Rails.logger.debug("user email: ", user.email)
```
Comment (victorhuangwq) [inline app/models/user.rb:144]: This was there before I edited it. Could you see if there's a way to log this without throwing the error?

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -572,6 +572,7 @@ def write_cuds(cuds)
                                                   grade_policy: new_cud[:grade_policy],
                                                   lecture: new_cud[:lecture])
         existing.assign_attributes(params.permit(:lecture, :section, :grade_policy))
+        existing.dropped = false
```
Comment (victorhuangwq) [inline app/controllers/courses_controller.rb:575]: Could you quickly explain what is happening by setting dropped = false? I assume this is for when dropped students are added back, they should become "undropped"

File: db/migrate/20210929220050_create_github_integrations.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@ def change
     create_table :github_integrations do |t|
       t.string :oauth_state
       t.text :access_token_ciphertext
-      t.references :user, foreign_key: true, index: { unique: true }
+      t.references :user, foreign_key: true, type: :integer, index: { unique: true }
```
Comment (victorhuangwq) [inline db/migrate/20210929220050_create_github_integrations.rb:6]: @fanpu this is the fix for the github integration issue btw

File: db/schema.rb [hunk 4]
```diff
@@ -12,7 +12,7 @@
 
 ActiveRecord::Schema.define(version: 2021_10_23_185328) do
```
Comment (victorhuangwq) [inline db/schema.rb:14]: @michellexliu I think you should un-checkin this commit, because you are currently using MySQL and it added more options

**Review by victorhuangwq — CHANGES_REQUESTED** [review]
Hi Michelle thanks for fixing the error so quickly. I have left my comments on each file, and once they are addressed, we should be good to go.

File: app/controllers/courses_controller.rb (hunk 2 above)
Comment (michellexliu) [reply app/controllers/courses_controller.rb:575]: Yep, that's correct! When we parse a "red" student, existing.dropped gets set to true, but previously we never set it back to false when the student gets added back in.

**Review by victorhuangwq — APPROVED** [review]
LGTM

**Pull request merged**